lib/layer.js

```diff
@@ -173,3 +173,15 @@ function decode_param(val){
     throw err
   }
 }
+
+function once(fn) {
+  var called = false
+
+  return function (err) {
+    if (!err) {
+      if (called) { throw new Error('next() cannot be called twice') }
```

Comment:
Wasn't entirely sure where you wanted the new line so I put it [here](https://github.com/calebmer/router/blob/many-next/lib/layer.js#L193).